device/isis/segmentrouting.yaml

```diff
@@ -0,0 +1,193 @@
+components:
+  schemas:
+    Isis.SegmentRouting:
+      description: |-
+        Segment Routing (SR) turns on any node to select any path (explicit or derived from IGPs SPT computations) 
+        for each of its traffic classes. The path does not depend on a hop-by-hop 
+        signaling technique (unlike LDP or RSVP). 
+        It only depends on a set of segments that are advertised by the IS-IS routing protocol. 
+        These segments act as topological sub-paths that can be combined together to form the required path.
+        Reference: https://datatracker.ietf.org/doc/html/rfc8667
+      type: object
+      properties:
+        router_capability:
+          description: |-
+            Optional IS-IS TLV named CAPABILITY, formed of multiple sub-TLVs, which allows a router to announce its
+            capabilities within an IS-IS level or the entire routing domain.        
+          $ref: '#/components/schemas/Isis.RouterCapability'
+          x-field-uid: 1
```

Comment:
So just a single attribute under this. Would things get added for SRv6 later here if needed ? Also need to evaluate at a top level if SRv6 will fit in well under router / interface and prefixes alongwith current proposal for MPLS SR.

Reply:
Both SR and SRv6 will be under Router capability. some configurations of SRv6 will be under Router Capability as they appeared sub-tlv to Router Capability tlv. Some configurations of Srv6 will be under SR-Capability as they are sub-tlv of SR-Capability sub-tlv of Router Capability tlv.  So, there is no conflict.  https://datatracker.ietf.org/doc/html/rfc9352#name-srv6-capabilities-sub-tlv https://datatracker.ietf.org/doc/html/[draft-ietf-isis-sr-yang-11](https://datatracker.ietf.org/doc/html/draft-ietf-isis-sr-yang-11)